packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1554,6 +1549,14 @@ class FocusManager with DiagnosticableTreeMixin, ChangeNotifier {
     rootScope._manager = this;
   }
 
+  // Used in the constructor and in [listenToApplicationLifecycleChangesIfSupported]
+  // to check if the app is running on a native mobile platform.
+  static final bool _mobileDevice = switch (defaultTargetPlatform) {
```

Comment:
I think we should probably add `kIsWeb` to this check, so if we ever do remove it, we only have to remove it from one place. Also consider moving the comments regarding Android/iOS support above this variable.